lint.go

```diff
@@ -1394,6 +1396,47 @@ func (f *file) lintTimeNames() {
 	})
 }
 
+func (f *file) lintDurationUnits() {
+	f.walk(func(node ast.Node) bool {
+		e, ok := node.(ast.Expr)
+		if !ok {
+			return true
+		}
+
+		tv := f.pkg.typesInfo.Types[e]
+		if tv.Type == nil || !f.pkg.isNamedType(tv.Type, "time", "Duration") {
+			return true
+		}
+		if tv.Value == nil {
+			return true
+		}
+		// Ok for 0 to be unitless; non-constants are ok
+		if x, ok := exact.Int64Val(tv.Value); !ok || x == 0 {
+			return true
+		}
+		if _, ok := e.(*ast.BasicLit); !ok {
```

Comment:
It is important that the earlier checks return true, whereas this returns false. I've expanded this a bit and given it a (hopefully helpful) comment. In the process, I also added handling of `time.Duration(30)`.